.github/workflows/release.yml

```diff
@@ -248,45 +266,32 @@ jobs:
     name: Publish pyogrio to GitHub / PyPI
     needs: [test-sdist, test-wheels]
     runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/pyogrio
+    permissions:
+      id-token: write # IMPORTANT: this permission is mandatory for trusted publishing to PyPI
+      contents: write # this permission is required for the Github release action
+
     # release on every tag
     if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/')
 
     steps:
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
-          name: artifact
+          pattern: pyogrio-*
           path: dist
+          merge-multiple: true
 
       - name: Publish distribution to PyPI
-        uses: pypa/gh-action-pypi-publish@master
-        with:
-          user: __token__
-          password: ${{ secrets.PYPI_API_TOKEN }}
+        uses: pypa/gh-action-pypi-publish@release/v1
 
       - name: Create GitHub Release
         id: create_release
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        uses: softprops/action-gh-release@v1
         with:
           tag_name: ${{ github.ref }}
-          release_name: ${{ github.ref }}
```

Comment:
In the email notification I got from your test, the title was "Release v0.7.3a3 - refs/tags/v0.7.3a3", while before this was something like "Release v0.7.2 - v0.7.2". I am not sure what the second part (that is now with "refs/tags/..") is used for in the UI (since you deleted it). Ideally this would still just be the version name, but this is of course also something easily edited manually (since we need to edit the release anyway to add some release notes)

Reply:
In the UI, it named it the same as the email, which is indeed not ideal.  We changed to a different action for creating the release, so that may have something to do with it.  I'll do a bit more experimenting to see if we can get it to name it nicely; will be a bit noisy with release announcement emails though!

Reply:
Turns out I was pointing it at the wrong variable previously, which is why it started including `refs/tags` prefix.

It is easy to add "Version <tag>" as the release; it does not appear easy to either use the message set on the tag per our instructions (e.g., "Version 0.7.3a3") nor does it appear easy to strip the leading "v" from the tag in order to create the name.  So now our release names by default are e.g., "Version v0.7.3a3".

This at least makes the messages sent out by github better and the default name of the version better; we can always edit it to remove the "v" when adding the changes, if we are that annoyed about the "v".